go-selinux/selinux_linux.go

```diff
@@ -1095,3 +1104,115 @@ func dupSecOpt(src string) ([]string, error) {
 func disableSecOpt() []string {
 	return []string{"disable"}
 }
+
+func getDefaultContextFromReaders(c defaultSECtx) (string, error) {
+	if c.verifier == nil {
+		return "", fmt.Errorf("verifier function must not be nil")
+	}
+
+	context, err := newContext(c.scon)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create label for %s", c.scon)
+	}
+
+	findUserInContext := func(r io.Reader) string {
+		fromRole := context["role"]
+		fromType := context["type"]
+		bufin := bufio.NewReader(r)
+
+		for done := false; !done; {
+			var (
+				line string
+				err  error
+			)
+
+			if line, err = bufin.ReadString('\n'); err != nil {
+				if err == io.EOF {
+					done = true
+				} else {
+					break
+				}
+			}
+			line = strings.TrimSpace(line)
+			if len(line) == 0 {
+				// Skip blank lines
+				continue
+			}
+			if line[0] == ';' || line[0] == '#' {
+				// Skip comments
+				continue
+			}
+
+			conns := strings.Fields(line)
+
+			lineConn := strings.TrimSpace(conns[0])
+			lineArr := strings.SplitN(lineConn, ":", 3)
+
+			// malformed context in the context file
+			if len(lineArr) != 3 {
+				continue
+			}
+
+			if lineArr[0] != fromRole || lineArr[1] != fromType {
+				continue
+			}
+
+			for _, cc := range conns[1:] {
+				userConn := strings.TrimSpace(cc)
+				userArr := strings.SplitN(userConn, ":", 3)
```

Comment:
1. This should be taken out of the `for` loop, otherwise you're doing the same thing again and again.

2. No check for a number of fields returned.

3. See https://github.com/opencontainers/selinux/pull/119/commits/6fd60bb08ac5d42cad52be3d3195e97464b57585#r497819835

Reply:
> This should be taken out of the for loop, otherwise you're doing the same thing again and again

Are you referring to the setting of user and level in the context or something else?

> See 6fd60bb#r497819835

This URL doesn't work for some reason. I'm assuming you meant setting the subslices in `SplitN` to 4?